packages/layout/src/Legend.ts

```diff
@@ -138,6 +138,25 @@ export class Legend extends SVGWidget {
         return Palette.ordinal("default");
     }
 
+    getPaletteType() {
+        return this.getPalette().type();
+    }
+
+    fillColorFunc() {
+        const widget = this.getWidget();
+        if (widget && widget.fillColor) {
+            return widget.fillColor;
+        }
+        const palette = Palette.ordinal("default");
```

Comment:
Why not use the widget's paletteID if it exists?
`const palette = Palette.ordinal(widget && widget.paletteID ? widget.paletteID() : "default");`

Reply:
Good point!